calicoctl/calicoctl.go

```diff
@@ -79,10 +81,11 @@ Options:
 		default:
 			fmt.Println(usage)
 		}
-	}
 
-	if err != nil {
-		fmt.Printf("Error executing command: %s\n", err)
-		os.Exit(1)
+		if err != nil {
+			fmt.Printf("Error executing command. Invalid option: 'calicoctl %s'. Use flag '--help' to read about a specific subcommand.\n", strings.Join(args, " "))
```

Comment:
What's the return code / error handling strategy for calicoctl?  If we're going to ignore the contents of the error here, we need to be sure that all reporting to the user is done inside the subcommands.